### DISCUSSION TITLE
the same dependency in all workflows / jobs

### DISCUSSION THREAD

**Discussion by kwladyka** [community/community#25805] [Actions]
<pre><code class="lang-auto">
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;runs-on:&amp;nbsp;ubuntu-18.04
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;steps:
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;-&amp;nbsp;name:&amp;nbsp;checkout
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;uses:&amp;nbsp;actions/checkout@v1
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;with:
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;ref: [this will be different]
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;fetch-depth:&amp;nbsp;1
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;token:&amp;nbsp;${{&amp;nbsp;secrets.GITHUB\_TOKEN&amp;nbsp;}}

&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;-&amp;nbsp;name:&amp;nbsp;Install java
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;uses:&amp;nbsp;actions/setup-java@v1
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;with:
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;java-version:&amp;nbsp;'13.0.1'  

&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;-&amp;nbsp;name:&amp;nbsp;Install clojure
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;uses:&amp;nbsp;DeLaGuardo/setup-clojure@2.0
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;with:
&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;tools-deps:&amp;nbsp;'1.10.1.478'  

&amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp;-&amp;nbsp;run:&amp;nbsp;npm i
</code></pre>
<ol>
<li>
<p>I want to start each workflow file in the same way. Do I have to create github action from that part in the same repository? No other way?</p>
</li>
<li>
<p>I would like to prepare all dependencies in job and later run a few jobs based on the state of the system after finish job1. Is it possible?</p>
</li>
</ol>
<ol>
<li>run job 1<br>
1.a. install java</li>
<li>run job 2, job 3, job 4 on system after finish job 1 (so with java installed)</li>
</ol>
<p>Do I have to install the same java version for each job separately? And more dependencies.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<ul>
<li><em>I want to start each workflow file in the same way. Do I have to create github action from that part in the same repository? No other way?</em></li>
</ul>
<p>There is not an actions template conception now, if you want to run a group of same actions in different workflows, you need to add those actions in each workflow file now.</p>
<ul>
<li><em>I would like to prepare all dependencies in job and later run a few jobs based on the state of the system after finish job1. Is it possible?</em></li>
</ul>
<p><em> </em></p>
<p>You can use <strong><a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idneeds" rel="nofollow noopener">jobs.&lt;job_id&gt;.needs</a></strong>. Identifies any jobs that must complete successfully before this job will run.</p>
<p>Example: </p>
<p><img src="https://user-images.githubusercontent.com/24806478/181093553-1ffe724f-fe75-4b69-a66f-392f31121f98.png" alt="job needs job.png" data-base62-sha1="7zbivuJf6ltlbOTJySSer4vkR2V" width="344" height="305"></p>
<p>For more information, you can refer to <a href="https://help.github.com/cn/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idneeds" rel="nofollow noopener">https://help.github.com/cn/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idneeds</a></p>
<ul>
<li><em>Do I have to install the same java version for each job separately? And more dependencies.</em></li>
</ul>
<p><em> </em></p>
<p>Yes, as jobs are separated, each job runs on different machine, you need to install java for each job.</p>

**Comment by kwladyka** [comment]
<p>Not what I wanted to hear, but true :slight_smile:</p>
<p>As workaround probably the best way is to create action in the same repository which preapre all dependencies for all workflows and jobs in the same way. Or just use 1 long job.</p>
